pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java

```diff
@@ -427,29 +498,161 @@ private static String getTaskType(String name) {
     return name.split(TASK_NAME_SEPARATOR)[1];
   }
 
-  public static class TaskCount {
-    private int _waiting;   // Number of tasks waiting to be scheduled on minions
-    private int _error;     // Number of tasks in error
-    private int _running;   // Number of tasks currently running in minions
-    private int _total;     // Total number of tasks in the batch
+  @JsonPropertyOrder({"taskState", "subTaskCount", "startTime", "executionStartTime", "subTaskInfos"})
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  public static class TaskDebugInfo {
+    private String _startTime;
+    private String _executionStartTime;
+    private TaskState _taskState;
+    private TaskCount _subTaskCount;
+    private List<TaskPartitionDebugInfo> _subTaskInfos;
 
-    public TaskCount() {
+    public TaskDebugInfo() {
+    }
+
+    public void setStartTime(String startTime) {
+      _startTime = startTime;
+    }
+
+    public void setExecutionStartTime(String executionStartTime) {
+      _executionStartTime = executionStartTime;
+    }
+
+    public void setTaskState(TaskState taskState) {
+      _taskState = taskState;
+    }
+
+    public void setSubTaskCount(TaskCount subTaskCount) {
+      _subTaskCount = subTaskCount;
+    }
+
+    public void addSubTaskInfo(TaskPartitionDebugInfo subTaskInfo) {
+      if (_subTaskInfos == null) {
+        _subTaskInfos = new ArrayList<>();
+      }
+      _subTaskInfos.add(subTaskInfo);
+    }
+
+    public String getStartTime() {
+      return _startTime;
+    }
+
+    public String getExecutionStartTime() {
+      return _executionStartTime;
+    }
+
+    public TaskState getTaskState() {
+      return _taskState;
     }
 
-    public void addToWaiting(int waiting) {
-      _waiting += waiting;
+    public TaskCount getSubTaskCount() {
+      return _subTaskCount;
     }
 
-    public void addToRunning(int running) {
-      _running += running;
+    public List<TaskPartitionDebugInfo> getSubTaskInfos() {
+      return _subTaskInfos;
+    }
+  }
+
+  @JsonPropertyOrder({"taskId", "state", "startTime", "finishTime", "participant", "info", "taskConfig"})
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  public static class TaskPartitionDebugInfo {
```

Comment:
Suggest renaming to `SubtaskDebugInfo` to match the json field name?

Reply:
Renamed TaskPartitionDebugInfo to SubtaskDebugInfo